src/NuGet.Clients/NuGet.PackageManagement.UI/PackageItemLoader.cs

```diff
@@ -329,5 +335,12 @@ public IEnumerable<PackageItemListViewModel> GetCurrent()
 
             return listItems.ToArray();
         }
+
+        public void Dispose()
+        {
+            _installedPackages = null;
+            _packageReferences = null;
+            _state.Dispose();
+        }
```

Comment:
If this class is never inherited, mark it as sealed, otherwise [implement the dispose pattern](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#implement-the-dispose-pattern)

Reply:
Reverted, I made this change at initial steps of removing deduplication. But cache was main culprit.